src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -560,6 +560,22 @@ public static void fixNmMdAndUq(final SAMRecord record, final ReferenceSequenceF
             record.setAttribute(SAMTag.NM.name(), SequenceUtil.calculateSamNmTag(record, referenceBases, 0, isBisulfiteSequence));
         }
         if (record.getBaseQualities() != SAMRecord.NULL_QUALS) {
+            calculateUq(record, refSeqWalker, isBisulfiteSequence);
+        }
+    }
+
+    /** Calculates and sets the UQ tag from the record and the reference
+     *
+     * @param record the record to be fixed
+     * @param refSeqWalker a ReferenceSequenceWalker that will be used to traverse the reference
+     * @param isBisulfiteSequence a flag indicating whether the sequence came from bisulfite-sequencing.
+     *
+     * No return value, modifies the provided record.
+     */
+    public static void calculateUq(final SAMRecord record, final ReferenceSequenceFileWalker refSeqWalker, final boolean isBisulfiteSequence) {
+        final byte[] referenceBases = refSeqWalker.get(refSeqWalker.getSequenceDictionary().getSequenceIndex(record.getReferenceName())).getBases();
```

Comment:
This should be moved inside the `if` to avoid unnecessary work when the `if` is false.